packages/salesforcedx-utils-vscode/src/telemetry/reporters/telemetryFile.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { appendFile } from 'fs';
+import { LOCAL_TELEMETRY_FILE } from '../../constants';
+import { TelemetryReporter } from '../interfaces';
+
+export class TelemetryFile implements TelemetryReporter {
+  appInsightsClient: undefined;
+  userOptIn: undefined;
+  logStream: undefined;
+
+  sendTelemetryEvent(
+    eventName: string,
+    properties?: { [key: string]: string } | undefined,
```

Comment:
optional types are `T | undefined` by default, can omit this `| undefined` if you want